newIDE/app/src/BehaviorsEditor/Editors/Physics2Editor/index.js

```diff
@@ -0,0 +1,586 @@
+// @flow
+import * as React from 'react';
+import { Line, Column } from '../../../UI/Grid';
+import Checkbox from 'material-ui/Checkbox';
+import SelectField from 'material-ui/SelectField';
+import MenuItem from 'material-ui/MenuItem';
+import SemiControlledTextField from '../../../UI/SemiControlledTextField';
+import ImagePreview from '../../../ResourcesList/ResourcePreview/ImagePreview';
+import ResourceSelector from '../../../ResourcesList/ResourceSelector';
+import ResourcesLoader from '../../../ResourcesLoader';
+import ShapePreview from './ShapePreview.js';
+import PolygonEditor from './PolygonEditor.js';
+import { type BehaviorEditorProps } from '../BehaviorEditorProps.flow';
+
+type Props = BehaviorEditorProps;
+
+type State = {|
+  image: string,
+  imageWidth: number,
+  imageHeight: number,
+|};
+
+function NumericProperty(props: {|
+  properties: gdMapStringPropertyDescriptor,
+  propertyName: string,
+  step: number,
+  onUpdate: (newValue: string) => void,
+|}) {
+  const { properties, propertyName, step, onUpdate } = props;
+
+  return (
+    <SemiControlledTextField
+      value={properties.get(propertyName).getValue()}
+      key={propertyName}
+      floatingLabelText={properties.get(propertyName).getLabel()}
+      floatingLabelFixed
+      step={step}
+      onChange={onUpdate}
+      type="number"
+    />
+  );
+}
+
+function BitProperty(props: {|
+  enabled: boolean,
+  propertyName: string,
+  pos: number,
+  spacing: boolean,
+  onUpdate: (enabled: boolean) => void,
+|}) {
+  const { enabled, propertyName, pos, spacing, onUpdate } = props;
+
+  return (
+    <div style={{ width: spacing ? '7.5%' : '5%' }}>
+      {
+        <Checkbox
+          checked={enabled}
+          onCheck={(e, checked) => onUpdate(checked)}
+        />
+      }
+      {spacing && (
+        <div
+          style={{ width: '33%' }}
+          key={propertyName + '-space' + pos.toString(10)}
+        />
+      )}
+    </div>
+  );
+}
+
+export default class Physics2Editor extends React.Component<Props, State> {
+  resourcesLoader: typeof ResourcesLoader;
+
+  constructor(props: Props) {
+    super(props);
+
+    this.resourcesLoader = ResourcesLoader;

```

Comment:
In fact, I think we can entirely avoid a constructor by doing:

```
      resourcesLoader = ResourcesLoader;
 
      state = {
       image: '',
       imageWidth: 0,
       imageHeight: 0,
     };
```

Reply:
Haha, something I would love to change, readed the constructor was the recommend way when you had default state values, but I trust you :)